src/test/e2e/services/group-service.e2e.test.ts

```diff
@@ -97,3 +97,31 @@ test('should not remove user from no SSO definition group', async () => {
     expect(groups.length).toBe(1);
     expect(groups[0].name).toEqual('no_mapping_group');
 });
+
+test('adding a root role to a group with a project role should fail', async () => {
+    const group = await groupStore.create({
+        name: 'root_group',
+        description: 'root_group',
+    });
+
+    stores.accessStore.addGroupToRole(group.id, 1, 'test', 'default');
+
+    try {
+        await groupService.updateGroup(
+            {
+                id: group.id,
+                name: group.name,
+                users: [],
+                rootRole: 1,
+                createdAt: new Date(),
+                createdBy: 'test',
+            },
+            'test',
+        );
+    } catch (any) {
+        expect(any.message).toContain(
+            'This group already has a project role and cannot also be given a root role',
+        );
+    }
+    expect.assertions(1);
```

Comment:
I believe there's a better way of doing this: https://github.com/Unleash/unleash/blob/bd53193dc95d16ec13e147070c7fb1071f8911f3/src/test/e2e/api/client/segment.e2e.test.ts#L469-L476 but I'm not picky with tests

Reply:
Outside of being fewer lines is it actually better (genuinely asking, I wanna know if this does something cool I'm not aware of)? Honestly I'm trying to avoid these goopy mixed await/promise chains because we keep on getting them wrong in subtle ways and so tired of our tests doing weird things

Reply:
...that being said, I see we do this in a lot of places. I've taken the change, gonna merge it